internal/command/testdata/test/ephemeral_resource/main.tftest.hcl

```diff
@@ -0,0 +1,6 @@
+run "validate_ephemeral_resource" {
+  assert {
+    condition = ephemeral.test_ephemeral_resource.data.value == "bar"
+    error_message = "We expect this to fail since ephemeral resources should be closed when this is evaluated"
```

Comment:
This is a reasonable solution for now but it is admittedly something I did not realise during the early research phase.

I think it would be interesting if `terraform test` in the future had a way of inspecting the in-memory state of ephemeral resources somehow. That will likely take some more planning and more work though.

Reply:
It's not really in-memory state that is the problem. ephemeral resources are closed as part of their operation at some point, but terraform test never runs within a plan / apply but always after so they are closed ahead of time. As they are not live at that point anymore (even if the value would have been still valid) we can't access the value here

Reply:
Understood, I just remember that one of the concerns we had was e.g. How does one know that a shape of an ephemeral value is what's expected, if it is never exposed to the user?

IMO `terraform test` is what users will naturally try to use in that context.

So in the absence of us satisfying that use case somehow with testing we'll need to build some separate debugging workflows later.

Again - not suggesting any changes to your PR here - just sharing the "bigger picture" in my mind.